lib/tools/ui/widgets/text_entry.dart

```diff
@@ -77,26 +77,33 @@ class TextEntry extends StatelessWidget {
         ),
       ),
       validator: (value) {
-        if (canBeEmpty) {
-          if (value == null || value.isEmpty) {
+        if (value == null || value.isEmpty) {
+          if (canBeEmpty) {
             return null;
+          } else {
+            return noValueError;
```

Comment:
I think that you are recreating the issue I intend to solve by passing `validator?.call(value);` here

Reply:
`validator?.call(value)` is called at the end of the validator function.